src/app/dataset-view/additional-components/dataset-settings-component/tabs/webhooks/dataset-settings-webhooks-tab.component.ts

```diff
@@ -0,0 +1,209 @@
+    styleUrls: ["./dataset-settings-webhooks-tab.component.scss"],
+    changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class DatasetSettingsWebhooksTabComponent extends BaseComponent implements OnInit {
+    @Input(RoutingResolvers.DATASET_SETTINGS_WEBHOOKS_KEY) public webhooksViewData: DatasetViewData;
+    public dataSource = new MatTableDataSource();
+
+    public isLoaded: boolean = false;
+    public readonly DISPLAY_COLUMNS: string[] = ["event", "status", "actions"];
+    public readonly WebhookSubscriptionStatus: typeof WebhookSubscriptionStatus = WebhookSubscriptionStatus;
+
+    private ngbModalService = inject(NgbModal);
+    private datasetWebhooksService = inject(DatasetWebhooksService);
+    private cdr = inject(ChangeDetectorRef);
+    private modalService = inject(ModalService);
+
+    public ngOnInit(): void {
+        this.updateTable();
+    }
+
+    private updateTable(): void {
+        this.datasetWebhooksService
+            .datasetWebhookSubscriptions(this.datasetBasics.id)
+            .pipe(
+                finalize(() => {
+                    this.isLoaded = true;
+                    this.cdr.detectChanges();
+                }),
+                takeUntilDestroyed(this.destroyRef),
+            )
+
+            .subscribe((data: WebhookSubscription[]) => {
+                this.dataSource.data = data;
+                this.cdr.detectChanges();
+            });
+    }
+
+    public get datasetBasics(): DatasetBasicsFragment {
+        return this.webhooksViewData.datasetBasics;
+    }
+
+    public createWebhook(): void {
+        const modalRef = this.ngbModalService.open(CreateSubscriptionModalComponent);
+        const modalRefInstance = modalRef.componentInstance as CreateSubscriptionModalComponent;
+        modalRefInstance.datasetBasics = this.webhooksViewData.datasetBasics;
+
+        from(modalRef.result)
+            .pipe(
+                switchMap((result: WebhookSubscriptionModalActionResult) =>
+                    result.payload
+                        ? this.datasetWebhooksService.datasetWebhookCreateSubscription(
+                              this.datasetBasics.id,
+                              result.payload,
+                          )
+                        : of(),
+                ),
+                takeUntilDestroyed(this.destroyRef),
+            )
+            .subscribe((data: CreateWebhookSubscriptionSucces | null) => {
+                if (data) {
+                    const modalRef = this.ngbModalService.open(CreateSubscriptionModalComponent, {
+                        backdrop: "static",
+                        keyboard: false,
+                    });
+                    const modalRefInstance = modalRef.componentInstance as CreateSubscriptionModalComponent;
+                    modalRefInstance.datasetBasics = this.webhooksViewData.datasetBasics;
+                    modalRefInstance.subscriptionData = data;
```

Comment:
I don't fully understand it. You've created a webhook and reopening the modal immediately?